app/Repositories/Mark/MarkRepository.php

```diff
@@ -99,59 +92,61 @@ public function getSubCumAvg($tex3, $st_id, $sub_id, $course_id, $year)
         $total = $tex1 + $tex2 + $tex3;
 
         return ($total > 0) ? round($total/$count, 1) : 0;
+
     }
 
     public function getSubjectMark($exam, $course_id, $sub_id, $st_id, $year)
     {
+        // TODO: Implement getSubjectMark() method.
         $d = [ 'exam_id' => $exam->id, 'my_course_id' => $course_id, 'subject_id' => $sub_id, 'student_id' => $st_id, 'year' => $year ];
         $tex = 'tex'.$exam->term;
 
-        return Mark::where($d)->select($tex)->get()->first()->$tex;
+        return $this->model->where($d)->select($tex)->get()->first()->$tex;
     }
 
     public function getSubPos($st_id, $exam, $course_id, $sub_id, $year)
     {
+        // TODO: Implement getSubPos() method.
         $d = ['exam_id' => $exam->id, 'my_course_id' => $course_id, 'subject_id' => $sub_id, 'year' => $year];
         $tex = 'tex'.$exam->term;
 
         $sub_mk = $this->getSubjectMark($exam, $course_id, $sub_id, $st_id, $year);
 
-        $sub_mks = Mark::where($d)->whereNotNull($tex)->orderBy($tex, 'DESC')->select($tex)->get()->pluck($tex);
+        $sub_mks = $this->model->where($d)->whereNotNull($tex)->orderBy($tex, 'DESC')->select($tex)->get()->pluck($tex);
         return $sub_pos = $sub_mks->count() > 0 ? $sub_mks->search($sub_mk) + 1 : NULL;
     }
 
     public function countExSubjects($exam, $st_id, $course_id, $year)
     {
+        // TODO: Implement countExSubjects() method.
         $d = [ 'exam_id' => $exam->id, 'my_course_id' => $course_id, 'student_id' => $st_id, 'year' => $year ];
         $tex = 'tex'.$exam->term;
 
         if($exam->term == 3){ unset($d['exam_id']); }
 
-        return Mark::where($d)->whereNotNull($tex)->count();
+        return $this->model->where($d)->whereNotNull($tex)->count();
     }
 
     public function getClassAvg($exam, $course_id, $year)
     {
+        // TODO: Implement getClassAvg() method.
         $d = [ 'exam_id' => $exam->id, 'my_course_id' => $course_id, 'year' => $year ];
         $tex = 'tex'.$exam->term;
 
-        $avg = Mark::where($d)->select($tex)->avg($tex);
+        $avg = $this->model->where($d)->select($tex)->avg($tex);
         return round($avg, 1);
     }
 
     public function getPos($st_id, $exam, $course_id, $sec_id, $year)
     {
+        // TODO: Implement getPos() method.
         $d = ['student_id' => $st_id, 'exam_id' => $exam->id, 'my_course_id' => $course_id, 'class_id' => $sec_id, 'year' => $year ]; $all_mks = [];
         $tex = 'tex'.$exam->term;
 
-        $my_mk = Mark::where($d)->select($tex)->sum($tex);
-
-        /*if($exam->term == 3){
-            $my_mk = Mark::where($d)->select('cum')->sum('cum');
-        }*/
+        $my_mk = $this->model->where($d)->select($tex)->sum($tex);
 
         unset($d['student_id']);
-        $mk = Mark::where($d);
+        $mk = $this->model->where($d);
```

Comment:
Anh không rõ mục đích em đặt biến $mk ở đây để làm gì (Chưa kể tên biến mk là cái gì đã rất khó hiểu rồi)
$mk = $this->model->where($d);
$students = $mk->select('student_id')->distinct()->get();

Đoạn trên em viết thành như này thì anh có thấy logic bị sai gì đâu nhỉ
$students = $this->model->select('student_id')->where($d)->distinct()->get();